### PR TITLE
chore(): Update modules deps

### PR DESCRIPTION
**Pull request by adrien2p**
RESOLVES FRMW-2716, FRMW-2715

**What**
Migrate all modules to use framework as a peer deps and update their usage

### CODE REVIEW

File: packages/core/framework/package.json [hunk 1]
```diff
@@ -23,15 +23,16 @@
...
+    "./modules-sdk": "./dist/modules-sdk/index.js"
   },
   "engines": {
     "node": ">=20"
   },
   "repository": {
     "type": "git",
     "url": "https://github.com/medusajs/medusa",
-    "directory": "packages/framework/framework"
+    "directory": "packages/framework"
```
Comment (carlos-r-l-rodrigues) [inline packages/core/framework/package.json:35]:
```suggestion
    "directory": "packages/core/framework"
```

**Review by carlos-r-l-rodrigues — APPROVED** [review]

**Review by thetutlage — DISMISSED** [review]

**Comment by adrien2p** [comment]
@olivermrbl would you have time to review that one please

**Comment by olivermrbl** [comment]
If all of you three have gone through it, we should be good 👍 

The goal with the PR makes sense.

**Review by thetutlage — APPROVED** [review]

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**